src/auth/userSession.js

```diff
@@ -0,0 +1,332 @@
+/* @flow */
+import queryString from 'query-string'
+import { AppConfig } from './appConfig'
+import type { SessionOptions } from './sessionData'
+import {
+  LocalStorageStore,
+  SessionDataStore,
+  InstanceDataStore
+} from './sessionStore'
+import {
+  redirectToSignInImpl,
+  redirectToSignInWithAuthRequestImpl,
+  handlePendingSignInImpl,
+  loadUserDataImpl
+} from './authApp'
+
+import {
+  makeAuthRequestImpl,
+  generateTransitKey
+} from './authMessages'
+
+import {
+  decryptContentImpl,
+  encryptContentImpl,
+  getFileImpl,
+  putFileImpl,
+  listFilesImpl
+} from '../storage'
+
+import type {
+  PutFileOptions
+} from '../storage'
+
+import {
+  nextHour
+} from '../utils'
+import {
+  MissingParameterError,
+  InvalidStateError
+} from '../errors'
+import { Logger } from '../logger'
+
+/**
+ * Represents an instance of a signed in user for a particular app.
+ *
+ * A signed in user has access to two major pieces of information
+ * about the user, the user's private key for that app and the location
+ * of the user's gaia storage bucket for the app.
+ *
+ * A user can be signed in either directly through the interactive
+ * sign in process or by directly providing the app private key.
+ * @type {UserSession}
+ */
+export class UserSession {
+  appConfig: AppConfig
+
+  store: SessionDataStore
+
+  constructor(options: {appConfig?: AppConfig,
+    sessionStore?: SessionDataStore,
+    sessionOptions?: SessionOptions }) {
+    let runningInBrowser = true
+
+    if (typeof window === 'undefined') {
+      runningInBrowser = false
+    }
+
+    Logger.debug(`UserSession: runningInBrowser: ${runningInBrowser ? 'yes' : 'no'}`)
```

Comment:
This is pretty verbose, as most apps will have this called on every page. I'd suggest only logging this if `runningInBrowser` is `false`.